app/routes.js

```diff
@@ -0,0 +1,173 @@
+import Signup from './pages/signup';
+import Profile from './pages/profile';
+import Patient from './pages/patient';
+import PatientNew from './pages/patientnew';
+import PatientData from './pages/patientdata';
+import RequestPasswordReset from './pages/passwordreset/request';
+import ConfirmPasswordReset from './pages/passwordreset/confirm';
+import EmailVerification from './pages/emailverification';
+
+import personUtils from './core/personutils';
+
+/**
+ * This function redirects any requests that land on pages that should only be
+ * visible when logged in if the user is logged out
+ * 
+ * @param  {Object} nextState
+ * @param  {Function} replaceState
+ *
+ * @return {boolean|null} returns true if hash mapping happened
+ */
+export const requireAuth = (api) => (nextState, replaceState) => {
+  if (!api.user.isAuthenticated()) {
+    replaceState(null, '/login');
+  }
+};
+
+/**
+ * This function redirects any requests that land on pages that should only be
+ * visible when no data storage is set up if the user has data storage set up
+ *
+ * @param  {Object} nextState
+ * @param  {Function} replaceState
+ *
+ * @return {boolean|null} returns true if hash mapping happened
+ */
+export const requireAuthAndNoPatient = (api, cb) => (nextState, replaceState) => {
+  if (!api.user.isAuthenticated()) {
+    replaceState(null, '/login');
+    cb();
+  }
+  else {
+    api.user.get(function(err, user) {
+      if (personUtils.isPatient(user)) {
+        replaceState(null, '/patients');
+        cb();
+      }
+      cb();
+    });
+  }
+};
+
+/**
+ * This function redirects any requests that land on pages that should only be
+ * visible when logged out if the user is logged in
+ * 
+ * @param  {Object} nextState
+ * @param  {Function} replaceState
+ *
+ * @return {boolean|null} returns true if hash mapping happened
+ */
+export const requireNoAuth = (api) => (nextState, replaceState) => {
+  if (api.user.isAuthenticated()) {
+    replaceState(null, '/patients');
+  }
+};
+
+/**
+ * This function redirects any requests that land on pages that should only be
+ * visible when the user hasn't yet verified their sign-up e-mail
+ * if the user already has completed the e-mail verification
+ *
+ * @param  {Object} nextState
+ * @param  {Function} replaceState
+ *
+ * @return {boolean|null} returns true if hash mapping happened
+ */
+export const requireNotVerified = (api, cb) => (nextState, replaceState) => {
+  api.user.get(function(err, user) {
+    if (err) {
+      // we expect a 401 Unauthorized when navigating to /email-verification
+      // when not logged in (e.g., in a new tab after initial sign-up)
+      if (err.status === 401) {
+        return cb();
+      }
+      return cb(err);
+    }
+    if (user.emailVerified === true) {
+      replaceState(null, '/patients');
+      return cb();
+    }
+    // we log the user out so that requireNoAuth will work properly
+    // when they try to log in
+    api.user.logout(() => {
```

Comment:
This is basically only relevant when:
- working with the entire stack locally using a `+skip` e-mail address to avoid ever doing verification
- and thus clicking Login directly from the /email-verification page